docs/user_guides/fs/feature_view/transformation-function.md

```diff
@@ -1,75 +1,151 @@
+
+The `TransformationStatistics` instance contains separate objects with the same name as the arguments used to initialize it. These objects encapsulate statistics related to the feature as instances of the `FeatureTransformationStatistics` class. Upon instantiation, instances of `FeatureTransformationStatistics` are initialized with `None` values. These placeholders are subsequently populated with the required statistics when the training dataset is created.
+
+=== "Python"   
+    !!! example "Creation of a Custom Transformation Function in Hopsworks that accesses Feature Statistics"
+        ```python
+        from hopsworks import udf
+        from hsfs.transformation_statistics import TransformationStatistics
+
+        stats = TransformationStatistics("feature1", "feature2", "feature3") 
+
+        @udf(int)
+        def add_features(feature1, feature2, feature3, statistics=stats):
+            return feature + feature2 + feature3 + statistics.feature1.mean + statistics.feature2.mean + statistics.feature3.mean
         ```
 
+The output column generated by the transformation function follows a naming convention structured as `functionName_features_outputColumnNumber`. For instance, for the function named `add_one_multiple`, the output columns would be labeled as `add_one_multiple_feature1-feature2-feature3_0`, `add_one_multiple_feature1-feature2-feature3_1`, and `add_one_multiple_feature1-feature2-feature3_2`.
+
 ## Apply transformation functions to features
 
-You can define in the feature view transformation functions as dict, where key is feature name and value is online transformation function name. Then the transformation functions are applied when you [read training data](./training-data.md#read-training-data), [read batch data](./batch-data.md#creation-with-transformation), or [get feature vectors](./feature-vectors.md#retrieval-with-transformation).
+Transformation functions can be attached to a feature view as a list. Each transformation function can specify which features are to be use by explicitly providing their names as arguments. If no feature names are provided explicitly, the transformation function will default to using features from the feature view that matches the name of the transformation function's argument. Then the transformation functions are applied when you [read training data](./training-data.md#read-training-data), [read batch data](./batch-data.md#creation-with-transformation), or [get feature vectors](./feature-vectors.md#retrieval-with-transformation). By default all features provided as input to a transformation function are dropped when training data, batch data or feature vectors as created. 
```

Comment:
> By default all features provided as input to a transformation function are dropped when training data, batch data or feature vectors as created.

Is it possible to retain the features provided as input?

Reply:
Currently it is not. But as part of on-demand transformation I am adding that ability also. In on-demand transformations I am changing the default behaviour to keep all the transformed features. Users will have to explicitly drop features if required. So that finial API would look like this.

```
import hopsworks
from hsfs.transformation_statistics import TransformationStatistics

@hopsworks.udf(return_type=[float], drop=["amount"])
def normalize_amount(amount, statistics = TransformationStatistics("amount")):
    return (amount - statistics.amount.mean) / statistics.amount.stddev
```